packages/typescript-client/src/client.ts

```diff
@@ -66,7 +66,13 @@ type ReservedParamKeys =
 
 type ParamsRecord = Omit<Record<string, string>, ReservedParamKeys>
 
-type ShapeStreamErrorHandler = (error: Error) => void
+type RetryOpts = {
+  params?: ParamsRecord
+  headers?: Record<string, string>
+}
+type ShapeStreamErrorHandler = (
+  error: Error
+) => void | Promise<void> | RetryOpts | Promise<RetryOpts>
```

Comment:
```suggestion
) => void | RetryOpts | Promise<void | RetryOpts>
```
nit, I assume it would accept a function returning e.g. `Promise<void>` - if not ignore me